tests/data/test_data_viz.py

```diff
@@ -0,0 +1,136 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, List, Sequence
+
+import numpy as np
+import torch
+from PIL import Image
+from pytorch_lightning import seed_everything
+from pytorch_lightning.trainer.states import RunningStage
+
+from flash.data.base_viz import BaseViz
+from flash.data.utils import _STAGES_PREFIX
+from flash.vision import ImageClassificationData
+
+
+def _rand_image():
+    return Image.fromarray(np.random.randint(0, 255, (196, 196, 3), dtype="uint8"))
+
+
+def test_base_viz(tmpdir):
+
+    seed_everything(42)
+    tmpdir = Path(tmpdir)
+
+    (tmpdir / "a").mkdir()
+    (tmpdir / "b").mkdir()
+    _rand_image().save(tmpdir / "a" / "a_1.png")
+    _rand_image().save(tmpdir / "a" / "a_2.png")
+
+    _rand_image().save(tmpdir / "b" / "a_1.png")
+    _rand_image().save(tmpdir / "b" / "a_2.png")
+
+    class CustomBaseViz(BaseViz):
```

Comment:
Can you test with a mock so we also test the order and argument inputs? See for example:

https://github.com/PyTorchLightning/pytorch-lightning/blob/127c52af747066b184faed215ab009ea6d77ad21/tests/callbacks/test_callbacks.py#L26

Reply:
Done !